tests/integration/requests_tests.py

```diff
@@ -157,3 +170,91 @@ def test_external_client_doesnt_send_headers(http_server):
         assert "X-Parent" not in http_server.requests[0].headers
         assert "X-Span" not in http_server.requests[0].headers
         assert "X-Edge-Request" not in http_server.requests[0].headers
+
+
+@pytest.mark.parametrize("client_cls", [InternalRequestsClient, ExternalRequestsClient])
+@pytest.mark.parametrize("client_name", [None, "", "complex.client$name"])
+@pytest.mark.parametrize("method", ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "PUT", "POST"])
+def test_prom(client_cls, client_name, method, http_server):
+    baseplate = Baseplate(
+        {"myclient.filter.ip_allowlist": "127.0.0.0/8", "myclient.filter.port_denylist": "0"}
+    )
+    logger.info("Created baseplate.")
+    if client_name is None:
+        baseplate.configure_context({"myclient": client_cls()})
+    else:
+        baseplate.configure_context({"myclient": client_cls(client_name=client_name)})
+    logger.info(
+        f"Configured baseplate context with requests client. [type={client_cls.__name__}, client_name={client_name}]"
+    )
+
+    # observer = TestBaseplateObserver()
+    # observer = TestPrometheusBaseplateObserver()
+    observer = PrometheusBaseplateObserver()
+    logger.info(f"Created observer. [type={observer.__class__.__name__}]")
+    baseplate.register(observer)
+    logger.info("Registered observer with baseplate.")
+
+    # we need to clear metrics between test runs from the parametrize annotations
+    REGISTRY._names_to_collectors["http_client_requests_total"].clear()
+    REGISTRY._names_to_collectors["http_client_latency_seconds"].clear()
+    REGISTRY._names_to_collectors["http_client_active_requests"].clear()
```

Comment:
While we have the getter methods, might as well use them?

Reply:
I don't have access to the Metrics class here.
The way we achieve this in other tests is by by creating test classes where we override a lot of things to be able to keep track of the child spans and then access their `.metrics` class variable.
Which IMO is gross, and not warranted here, since I can just query the registry.
It also lets me ensure that we don't have some rogue metrics created that we lose track of.